src/qt/ee_debugwindow.hpp

```diff
@@ -0,0 +1,178 @@
+#ifndef EE_DEBUGWINDOW_H
+#define EE_DEBUGWINDOW_H
+
+#include <QWidget>
+#include <QtCore/QVariant>
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QComboBox>
+#include <QtWidgets/QGridLayout>
+#include <QtWidgets/QGroupBox>
+#include <QtWidgets/QHeaderView>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QLineEdit>
+#include <QtWidgets/QPushButton>
+#include <QtWidgets/QSpacerItem>
+#include <QtWidgets/QTableWidget>
+#include <QtWidgets/QWidget>
+#include "../core/ee/emotion_breakpoint.hpp"
+
+enum class BreakpointType
+{
+    AND,
+    MEMORY,
+    REGISTER,
+    PC,
+    STEP
+};
+
+class EmuThread;
+
+class EEDebugWindow : public QWidget, public EmotionDebugUI
+{
+
+    enum class DebuggerCpu
+    {
+        EE,
+        IOP
+    };
+
+    Q_OBJECT
+
+public:
+    explicit EEDebugWindow(EmuThread* emu_thread, QWidget *parent = 0);
```

Comment:
To be explicit, `nullptr` should be used here, unless there is a specific reason 0 is being used?

```suggestion
    explicit EEDebugWindow(EmuThread* emu_thread, QWidget *parent = nullptr);
```